src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -973,6 +983,121 @@ public void uninstall_noop(ChocolateyConfiguration config)
             return packageUninstalls;
         }
 
+        private void validate_package_names(ChocolateyConfiguration config)
+        {
+            foreach (var packageName in config.PackageNames.Split(';'))
+            {
+                if (packageName.EndsWith(NuGetConstants.PackageExtension))
+                {
+                    if (Uri.TryCreate(packageName, UriKind.Absolute, out var uri) && (uri.IsFile || uri.IsUnc))
+                    {
+                        throw_invalid_path_used(uri.LocalPath, uri.IsUnc, config.CommandName);
+                    }
+                    else if (_fileSystem.file_exists(packageName))
+                    {
+                        var fullPath = _fileSystem.get_full_path(packageName);
+
+                        if (!string.IsNullOrWhiteSpace(fullPath) && Uri.TryCreate(fullPath, UriKind.Absolute, out uri))
+                        {
+                            throw_invalid_path_used(uri.LocalPath, uri.IsUnc, config.CommandName);
+                        }
+
+                        throw_invalid_path_used(fullPath, isUncPath: false, commandName: config.CommandName);
+                    }
+                    else
+                    {
+                        throw new ApplicationException("Package name cannot point directly to a local or remote file. Please use the --source argument and point it to a local file directory, UNC directory path or a NuGet feed instead.");
+                    }
+                }
+                else if (packageName.EndsWith(NuGetConstants.ManifestExtension))
+                {
+                    throw new ApplicationException("Package name cannot point directly to a package manifest file. Please create a package first by running 'choco pack' on the .nuspec file first.");
+                }
+            }
+        }
+
+        private void throw_invalid_path_used(string packageName, bool isUncPath, string commandName)
+        {
+            var sb = new StringBuilder("Package name cannot be a path to a file ");
+
+            if (isUncPath)
+            {
+                sb.AppendLine("on a UNC location.")
+                  .AppendLine()
+                  .AppendLine("To install or upgrade a file in a UNC location, you may use:");
+            }
+            else
+            {
+                sb.AppendLine("on a remote or local file system.")
+                  .AppendLine()
+                  .AppendLine("To install or upgrade a local or remote file, you may use:");

```

Comment:
```suggestion
                  .AppendLine("To install or upgrade a local, or remote file, you may use:");

```

Also, same command as above regarding being specific to the commandName being passed in.